scrapy/pipelines/media.py

```diff
@@ -139,6 +139,11 @@ def _cache_result_and_execute_waiters(self, result, fp, info):
             result.cleanFailure()
             result.frames = []
             result.stack = None
+
+            # See twisted.internet.defer.returnValue docstring
```

Comment:
Since there are no tests, at least a comment mentioning it fixes a memory leak

Reply:
The commit message has the same description as this pull request. 
I guess we're good checking the commit message for more details.

Reply:
`_DefGen_Return` is private and very inner detail of inlineCallbacks implementation, I think it deserves a long comment explaining what it tries to prevents.